### ISSUE TITLE
Fake data

### ISSUE THREAD

**Issue by peterdudfield** [openclimatefix/uk-pv-national-gsp-api#354]
## Detailed Description
It would be great to run the api with fake data. This means users could run this locally without having to connect to a database.
The FE users could then use this API with fake data.

## Context
- To run this API locally right now, you need access to OCF database

## Possible Implementation
- add FAKE as a envvionrment
- add if statement in each route, to produce fake data. 
- Usefuly to have a general solar profile a bit like [this](https://github.com/openclimatefix/india-api/blob/main/src/india_api/internal/inputs/dummydb/client.py#L126)
- Need to match return types for each route
- Could do one route at a time, doesnt need to be one big PR
- Might want to add some noise on the forecast, so the actual is exactly the same

**Comment by VikramsDataScience** [comment]
Hello there @peterdudfield. This definitely looks to be a more complex and interesting change. If possible I'd like to attempt a contribution to this change? 

If so, could I just clarify the following points, please (apologies if these questions are a bit dumb/obvious!):
- Would creating the new FAKE environment be under the [shared workflows repo](https://github.com/openclimatefix/.github)?
- Would the new fake data be a modification to the nowcasting repo? Namely the [fake.py module](https://github.com/openclimatefix/nowcasting_datamodel/blob/main/nowcasting_datamodel/fake.py)? Or something that's entirely new?
- Could you also clarify the statement regarding: 
> add if statement in each route, to produce fake data.

What is meant by the route (again, sorry if these are obvious/dumb questions to ask)?

> Need to match return types for each route 
- Would the return types be in reference to matching the dtypes that the `DummyDBPredictedPowerProduction()` class returns found in the [_models.py module?](https://github.com/openclimatefix/india-api/blob/main/src/india_api/internal/inputs/dummydb/_models.py)
> Might want to add some noise on the forecast, so the actual is exactly the same
- Looking at your implementation in the `_basicSolarPowerProductionFunc()` function in the [client.py](https://github.com/openclimatefix/india-api/blob/main/src/india_api/internal/inputs/dummydb/client.py#L178) this seems to address that problem? Unless you're looking to improve this function further? In which case I could potentially look to conduct some research on how that might be improved (for instance, Meta did quite a cool implementation of a fourier series for their Prophet algorithm to address seasonality that might prove useful)? I can't promise I'll be successful, but if you'd like, I can certainly give it a try! Please do let me know how you'd like me to proceed, and I'll do my best.

**Comment by peterdudfield** [comment]
This repo and [code](https://github.com/openclimatefix/pv-site-api/blob/main/pv_site_api/main.py#L199) is how I was thinking it could be done

**Comment by VikramsDataScience** [comment]
Apologies @peterdudfield! I've been a little time poor, and distracted lately. I erroneously created a PR (https://github.com/openclimatefix/india-api/pull/76) in the india-api repo to address this issue, but I think I made the changes to the incorrect repo! I've since closed it, because I think we're looking to make similar changes but to this repo? If so, which module should I be looking to modify in this repo?

**Comment by peterdudfield** [comment]
No problem. See point 4 above, but we should try to fake all the 'routes' of the api

**Comment by VikramsDataScience** [comment]
Hey @peterdudfield. I've made the changes to what I think are the correct modules, but I'm running python 3.11.5 on my local machine and venv, but it looks like the `.pre-commit-config.yaml` requires python 3.9? Is there any way around this, as I'm really not too keen on downgrading my existing python version?

**Comment by VikramsDataScience** [comment]
Yeah, exactly! It prevents the commit from going through. Are you okay with me modifying the `default_language_version` in the `.pre-commit-config.yaml`?

**Comment by peterdudfield** [comment]
yea, feel free too

**Comment by VikramsDataScience** [comment]
Cool. Thank you.

Now that I've got the changes made to the `.pre-commit-config.yaml` file, it looks like my changes are failing some of the pre-commit checks. It's a bit late here, but if its cool with you, I'll take a gander at resolving these issues over tomorrow/day after, and report back if I'm having any additional challenges? 

Otherwise, if all is good, I'll create a PR, and we can work together to get it right :).

**Comment by peterdudfield** [comment]
yea, of course,
Thanks so much for helping out on this

**Comment by VikramsDataScience** [comment]
Sorry @peterdudfield about all the commits! That was me having fun and playing around with the `pre-commit` library 😂

**Comment by VikramsDataScience** [comment]
Hey @peterdudfield. I've created the PR. It was initially failing the CI checks, but I've modified the `.pre-commit-config.yaml` again to use a more flexible python3 build over the python3.11.5. It looks like using the 3.11.5 was too narrow a declaration in the CI workflow, and was raising errors when trying to build the environment. 

I also quite foolishly left the modified relative imports that I used for testing in my push, not realising that the dockerfile already sets the PYTHONPATH.

The CI tests seem to be passing now. When you've got some time, could you review the changes in the PR, and let me know if I'm on the right track, please?

**Comment by Mohammad-Tamerr** [comment]
can you assign me?

**Comment by peterdudfield** [comment]
I think this is already done @Mohammad-Tamerr , thanks though. @VikramsDataScience can you confirm its done?

**Comment by VikramsDataScience** [comment]
Hey @peterdudfield. 

Yep. We merged the code into the main branch just before Christmas.
